rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -0,0 +1,258 @@
+The ops will be able to only given access to some of the common components when configuring the pipeline.
+The devs will have the ability to consume the provided components without being able to change the common configuration.
+The users will be able to monitor their pipelines through the `internal_metrics` source.
+
+## Motivation
+
+- Helps Vector to grow organically within an organization by allowing teams to adopt Vector at their own pace without heavy administrator involvement.
+- Reduces the management overhead of devops/SREs by enabling teams to manage their own pipelines (spread the management load).
+
+## Implementation
+
+### How and where the pipelines are stored
+
+To avoid backward incompatibility, pipelines will be loaded from a `pipelines` sub-directory relative to the Vector configuration directory (e.g., `/etc/vector/pipelines`). Therefore, if a user changes the location of the Vector configuration directory they will also change the `pipelines` directory path. They are coupled.
+The `pipelines` directory will contain all pipelines represented as individual files. For simplicity, and to ensure users do not over complicate pipeline management, sub-directories/nesting are not allowed. This is inspired by Terraform's single-level directory nesting, which has been a net positive for simple management of large Terraform projects.
+Each pipeline file is a processing subset of a larger Vector configuration file. Therefore, it follows the same syntax as Vector's configuration (`toml`, `yaml`, and `json`).
+
+### Configuration
+
+Each time Vector will read its configuration file (on boot and when it live reloads), it will first read the pipeline configuration files and associate an `id` attribute corresponding to the pipeline's filename.
+
+A pipeline can be referenced in a configuration file as follows
+
+```toml
+# vector.toml
+[resources.pipeline_id]
+type = "pipeline"
+filename = "pipeline_id.toml"
+[resources.pipeline_id.forwards]
+source_alias = ["nginx", "apache", "transform_name"]
+sink_alias = ["elastic", "blackhole"]
+```
+
+The filename is used to point out the pipeline configuration file in the `pipelines` folder.
+In the pipeline's configuration file, the `id` can be set manually.
+A `load-balancer.yml` will, by default, have `load-balancer` as `id`.
+If the `id` matches the name of the file, the `filename` field can be omitted in the pipeline's declaration.
+
+If several files having the same `id` in that folder (for example `load-balancer.yml` and `load-balancer.json`), vector should error.
+
+The pipeline's configuration files should only contain transforms or vector will error.
+A pipeline cannot use another pipeline or vector will error.
+
+A pipeline has access to the components configured in the `forwards` section.
+The forwards section in the pipeline declaration allows to create an alias for the resources that will be accessible to avoid name conflicts when using multiple vector configuration files.
```

Comment:
I would remove this for now. For the first version the pipeline has access to everything.